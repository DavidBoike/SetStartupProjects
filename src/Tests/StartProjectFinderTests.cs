using System.IO;
using System.Xml.Linq;
using NUnit.Framework;
using ObjectApproval;
using Resourcer;
using SetStartupProjects;

[TestFixture]
public class StartProjectFinderTests
{
    [Test]
    public void Exe_from_OutputType()
    {
        var projectText = Resource.AsString("OutputType_Exe.txt");
        Assert.IsTrue(new StartProjectFinder().ShouldIncludeProjectXml(XDocument.Parse(projectText),""));
    }

    [Test]
    public void WinExe_from_OutputType()
    {
        var projectText = Resource.AsString("OutputType_WinExe.txt");
        string guid;
        Assert.IsTrue(new StartProjectFinder().ShouldIncludeProjectXml(XDocument.Parse(projectText), ""));
    }

    [Test]
    public void StartActionIsProgram()
    {
        var projectText = Resource.AsString("StartActionIsProgram.txt");
        Assert.IsTrue(new StartProjectFinder().ShouldIncludeProjectXml(XDocument.Parse(projectText), ""));
    }

    [Test]
    public void Lib_from_OutputType()
    {
        var projectText = Resource.AsString("OutputType_Lib.txt");
        Assert.IsFalse(new StartProjectFinder().ShouldIncludeProjectXml(XDocument.Parse(projectText), ""));
    }

    [Test]
    public void Multiple_excluded_project_types()
    {
        var projectText = Resource.AsString("Multiple_Exclude.txt");
        Assert.IsFalse(new StartProjectFinder().ShouldIncludeProjectXml(XDocument.Parse(projectText), ""));
    }

    [Test]
    public void Multiple_include_project_types()
    {
        var projectText = Resource.AsString("Multiple_Include.txt");
        Assert.IsTrue(new StartProjectFinder().ShouldIncludeProjectXml(XDocument.Parse(projectText), ""));
    }

    [Test]
    public void Lower_project_types()
    {
        var projectText = Resource.AsString("Lower_Include.txt");
        Assert.IsTrue(new StartProjectFinder().ShouldIncludeProjectXml(XDocument.Parse(projectText), ""));
    }

    [Test]
    public void GetAllProjectFiles()
    {
        var samplesolutionTxt = Path.GetFullPath("SampleSolution.txt");
        var allProjectFiles = new StartProjectFinder().GetAllProjectFiles(samplesolutionTxt);
        var directoryName = Path.GetDirectoryName(samplesolutionTxt);
        ObjectApprover.VerifyWithJson(allProjectFiles,s => s.Replace(@"\\",@"\").Replace(directoryName,""));
    }
}