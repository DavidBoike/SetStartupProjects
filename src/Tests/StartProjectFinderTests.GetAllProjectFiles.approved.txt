[
  {
    "Guid": "99999999-9999-9999-9999-999999999999",
    "Path": "\ClassLibraryCS\ClassLibraryCS.csproj"
  },
  {
    "Guid": "11111111-1111-1111-1111-111111111111",
    "Path": "\ConsoleApplicationCS\ConsoleApplicationCS.csproj"
  },
  {
    "Guid": "628A29CC-E572-40C4-8AEE-333D73D07404",
    "Path": "\ClassLibraryVB\ClassLibraryVB.vbproj"
  },
  {
    "Guid": "97AE33BD-7F0D-40AC-9964-A771771885DC",
    "Path": "\ConsoleApplicationVB\ConsoleApplicationVB.vbproj"
  },
  {
    "Guid": "18606457-FCF0-4D4E-BC5C-FB11D1230409",
    "Path": "\ConsoleApplicationFS\ConsoleApplicationFS.fsproj"
  },
  {
    "Guid": "1372F482-BCD4-4DDE-9B20-2999936AEEA1",
    "Path": "\WpfApplicationVB\WpfApplicationVB.vbproj"
  },
  {
    "Guid": "59AAAFE1-198C-4BB4-9538-072F8A5DC9DF",
    "Path": "\WpfApplicationCS\WpfApplicationCS.csproj"
  },
  {
    "Guid": "A091F9A6-5AC1-4EFB-86B0-9A5EAA8C4A20",
    "Path": "\WindowsFormsApplicationCS\WindowsFormsApplicationCS.csproj"
  },
  {
    "Guid": "11A80128-D9CA-4D63-8525-43854CFBD9DB",
    "Path": "\WindowsFormsApplicationVB\WindowsFormsApplicationVB.vbproj"
  },
  {
    "Guid": "660A14D1-72D0-4A9D-A51B-D0BC338E7688",
    "Path": "\WebApplicationVB\WebApplicationVB.vbproj"
  },
  {
    "Guid": "F8D988CE-ED35-4703-958A-F7771E4AF722",
    "Path": "\WebApplicationCS\WebApplicationCS.csproj"
  },
  {
    "Guid": "915CBC1C-4F15-4293-B0C4-787CE4E82787",
    "Path": "\ClassLibraryFS\ClassLibraryFS.fsproj"
  },
  {
    "Guid": "C188FCEA-C469-4B03-B39E-DC092370F0EC",
    "Path": "\AzureCloudService\AzureCloudService.ccproj"
  }
]